src/engine.rs

```diff
@@ -1126,4 +1135,76 @@ mod tests {
             });
         }
     }
+
+    #[test]
+    fn test_dump_file_or_directory() {
+        let dir = tempfile::Builder::new()
+            .prefix("test_dump_file_or_directory")
+            .tempdir()
+            .unwrap();
+        let entry_data = vec![b'x'; 1024];
+
+        let mut batches = vec![vec![LogBatch::default()]];
+        let mut batch = LogBatch::default();
+        batch
+            .add_entries::<Entry>(7, &generate_entries(1, 11, Some(&entry_data)))
+            .unwrap();
+        batch.add_command(7, Command::Clean);
+        batch.put(7, b"key".to_vec(), b"value".to_vec());
+        batch.delete(7, b"key2".to_vec());
+        batches.push(vec![batch.clone()]);
+        let mut batch2 = LogBatch::default();
+        batch2.put(8, b"key3".to_vec(), b"value".to_vec());
+        batch2
+            .add_entries::<Entry>(8, &generate_entries(5, 15, Some(&entry_data)))
+            .unwrap();
+        batches.push(vec![batch, batch2]);
+
+        let cfg = Config {
+            dir: dir.path().to_str().unwrap().to_owned(),
+            ..Default::default()
+        };
+
+        let engine =
+            RaftLogEngine::open_with_file_builder(cfg, Arc::new(DefaultFileBuilder)).unwrap();
+        for bs in batches.iter_mut() {
+            for batch in bs.iter_mut() {
+                let _ = engine.write(batch, false);
+            }
+
+            let _ = engine.sync();
+        }
+
+        //dump dir with raft groups. 8 element with raft groups 7 and 2 elements with raft groups 8
+        let dump_it = Engine::dump(dir.path()).unwrap();
+        let mut total = 0;
+        for item in dump_it {
+            if item.is_ok() {
+                total += 1;
+            }
+        }
+        assert!(total == 10);
```

Comment:
We can use `dump_it.count()`

Reply:
Should we need to check the item is Ok? as iterator will return Ok(v) or Err(msg)?

Reply:
Oh, that makes sense. You can call `dump_it.map(|i| i.unwrap()).count()`.